test/unit/test_test_await_methods.gd

```diff
@@ -87,7 +87,34 @@ class TestTheNewWaitMethods:
 		await wait_for_signal(signaler.the_signal, 1)
 		assert_between(counter.time, .9, 1.1)
 
+	func test_assert_eventually_waits_until_predicate_function_returns_true():
+		var some_node = add_child_autoqfree(Node.new())
+		var is_named_foo = func(): return some_node.name == 'foo'
+		var signaler = add_child_autoqfree(TimedSignaler.new())
+		signaler.the_signal.connect(func(): some_node.name = 'foo')
+		signaler.emit_after(.1)
+
+		assert_false(is_named_foo.call())
+		await assert_eventually(is_named_foo, 1.0)
+		assert_true(is_named_foo.call())
+
+		assert_between(counter.time, .09, .15)
+
+	# TODO: Keep an eye on the Github issue, until Godot fixes the error log
+	# This test passes, but prints the error log:
+	# 	ERROR: Lambda capture at index 0 was freed. Passed "null" instead
+	# Godot issue: https://github.com/godotengine/godot/issues/85947
+	func test_wait_until_is_compatible_with_checking_if_an_object_is_freed():
```

Comment:
(can't speak to which one's better between `is_instance_valid `vs `is_freed`)

I think a high level test like this should only use what a user would be allowed to use.

That leaves 2 options:
- keep `is_instance_valid` for now. Hopefully the Godot team will fix what's causing the error messages
- expose `is_freed `in the public API, document, etc
  - (!) once public, it's tough to walk back, so make sure it fits in your vision
  - could be in separate PR
  - longer term it would be awesome to expose a bunch of helper predicate functions
    - Kinda like [Hamcrest matchers](https://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html) or [Jest matchers ](https://jestjs.io/docs/expect#matchers)

Either way I left a comment in the code so to help keep track of it